File: lib/models/landmark.dart
class Landmark {
  final int id;
  final String name;
  final String description;
  final double distance;
  final String imageUrl;
  final double overallRating;
  final double recentRating;
  final List<String> categories;

  Landmark({
    required this.id,
    required this.name,
    required this.description,
    required this.distance,
    required this.imageUrl,
    required this.overallRating,
    required this.recentRating,
    required this.categories,
  });

  factory Landmark.fromJson(Map<String, dynamic> json) {
    return Landmark(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      distance: (json['distance'] as num).toDouble(),
      imageUrl: json['imageUrl'],
      overallRating: (json['overallRating'] as num).toDouble(),
      recentRating: (json['recentRating'] as num).toDouble(),
      categories: List<String>.from(json['categories']),
    );
  }
}

File: SpotNowAPI/SpotNowAPI/Models/Landmark.cs
namespace SpotNowAPI.Models
{
    public class Landmark
    {
        public int id { get; set; }
        public required string name { get; set; }
        public required string description { get; set; }
        public double distance { get; set; }
        public required string imageUrl { get; set; }

        public double overallRating { get; set; }
        public double recentRating { get; set; }
        
        public required String[] categories { get; set; }
        public int viewsToday { get; set; }
    }
}
